File: examples/tentacles/save-text/handler.js
#!/usr/bin/env node
const os = require("os");
const path = require("path");
let raw = "";
process.stdin.on("data", (c) => { raw += c; });
process.stdin.on("end", () => {
  const req = JSON.parse(raw || "{}");
  const p = req.payload || {};
  const blob = p.result || {};
  const inner = blob.result && typeof blob.result === "object" ? blob.result : blob;
  const text = inner.value || inner.text || "";
  const dest = path.join(os.tmpdir(), "kraken-browse.txt");
  const patch = {
    kind: "file",
    value: dest,
    meta: { bytes: String(text).length, from: inner.kind || "text" },
  };
  process.stdout.write(JSON.stringify({
    v: 1,
    id: req.id,
    ok: true,
    result: { patch, target: p.target },
  }));
});

File: examples/tentacles/compile-stamp/handler.py
#!/usr/bin/env python3
import json
import sys


def main() -> int:
    req = json.loads(sys.stdin.read() or "{}")
    payload = req.get("payload") or {}
    raw = payload.get("result") or {}
    inner = raw.get("result") if isinstance(raw.get("result"), dict) else raw
    patch = {
        "toolchain": "c",
        "compiled": True,
        "hello": inner.get("hello") or inner.get("result", {}).get("hello"),
    }
    sys.stdout.write(
        json.dumps(
            {"v": 1, "id": req.get("id"), "ok": True, "result": {"patch": patch}}
        )
    )
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
